feature-libs/order/components/amend-order/returns/order-return.guard.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router, UrlTree } from '@angular/router';
+import { SemanticPathService } from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { OrderReturnService } from './order-return.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OrderReturnGuard implements CanActivate {
+  constructor(
+    protected orderAmendService: OrderReturnService,
+    protected semanticPathService: SemanticPathService,
+    protected router: Router
+  ) {}
+
+  canActivate(): Observable<boolean | UrlTree> {
+    return this.orderAmendService.getForm().pipe(
+      map((form) => {
+        if (!form.valid) {
+          // the order code is not available in the route
+          // as long as we're inside a guard, hence we redirect
+          // to the common orders page.
+          return this.router.parseUrl(this.semanticPathService.get('orders'));
+        } else {
+          return true;
+        }
+      })
+    );
+  }
+}
```

Comment:
General comment about guards in the `components` entry point: are they only applied on components?  If so It’s fine to leave them in the `components` entry point.  Otherewise, if they apply to routes I think they have to be in `root`.

Reply:
double checked, they are only applied to components.